src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilder.java

```diff
@@ -284,6 +293,48 @@ private boolean verify(
         kubectl, manifestObjects, consoleLogger, verifyTimeoutInMinutes);
   }
 
+  /**
+   * @param credentialsId The service account credential's id.
+   * @return Access token from OAuth to allow kubectl to interact with the cluster.
+   * @throws IOException If an error occurred fetching the access token.
+   */
+  public static String getAccessToken(String credentialsId) throws IOException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(credentialsId));
+    HttpTransport HTTP_TRANSPORT = new NetHttpTransport();
+    JsonFactory JSON_FACTORY = new JacksonFactory();
+
+    // get the credential from jenkins
+    List<DomainRequirement> domainRequirements = new ArrayList<>();
+
+    GoogleRobotPrivateKeyCredentials cred =
+        CredentialsMatchers.firstOrNull(
+            CredentialsProvider.lookupCredentials(
+                GoogleRobotPrivateKeyCredentials.class,
+                Jenkins.get(),
+                ACL.SYSTEM,
+                domainRequirements),
+            CredentialsMatchers.withId(credentialsId));
+
+    if (cred != null) {
```

Comment:
If cred is null then it's an error, should through an exception in that case.

Reply:
I wanted to throw an exception, but this method would be called via kubernetesenginebuilder perform, and that method only throws IOException or InterruptedException. Should I be throwing one of these while doing a SEVERE level log?

Reply:
An AbortException would be appropriate in this case.

Reply:
Taken care of with de-duplication effort.